File: src/main/java/org/seminify/application/Application.java
package org.seminify.application;

import org.seminify.application.file.FileProperties;
import org.seminify.application.file.FileService;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
@EnableConfigurationProperties(FileProperties.class)
public class Application {

  public static void main(String[] args) {
    SpringApplication.run(Application.class, args);
  }

  @Bean
  CommandLineRunner commandLineRunner(FileService fileService) {
    return args -> {
      fileService.deleteAll();
      fileService.init();
    };
  }
}

File: src/main/kotlin/org/seminify/application/Application.kt
package org.seminify.application

import org.seminify.application.file.FileProperties
import org.seminify.application.file.FileService
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean

@SpringBootApplication
@EnableConfigurationProperties(FileProperties::class)
class Application {
    @Bean
    fun commandLineRunner(fileService: FileService) = CommandLineRunner {
        fileService.deleteAll()
        fileService.init()
    }
}

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
